src/components/BounceDetailsContainer/index.test.jsx

```diff
@@ -15,102 +16,89 @@ const {
-  priority: 0,
-  description:
-    "bWFpbmx5IGxpYmVydHkgZG9tYWluIGJsb2NrIHNlZWluZyB+NTAlIG9mIGFkZHJlc3NlcyBlbmdhZ2luZyBTRyB3aWRl",
-  bounce_action: "no_action",
-  user_id: 0,
-};
-
-const props = {
-  currentRule: sampleRule,
-  changelog: [sampleRule],
-  filteredChangelog: [sampleRule],
-  isEditClicked: false,
-  isChangeModalOpen: false,
-  isCancelConfirmOpen: false,
-  isConfirmOpen: false,
-  isUpdateError: false,
-  pageIndex: 1,
-  pageInterval: 10,
-  pagesToDisplay: 5,
-  isNetworkError: false,
-  changelogLimit: 10,
-  handleModalClose: jest.fn(),
-  handleButtonClicked: jest.fn(),
-  onChangeRule: jest.fn(),
-  handleEditClicked: jest.fn(),
-  handleCancelSaveClicked: jest.fn(),
-  handleChangelogClicked: jest.fn(),
-  handleCancelConfirmation: jest.fn(),
-  handleSaveConfirmation: jest.fn(),
-  onChangeRuleInt: jest.fn(),
-  updatePageIndex: jest.fn(),
-};
-
-const BounceRuleDetailedObject = <BounceRuleDetailed {...props} />;
+const [sampleRule] = mockBounceRules;
+const [sampleChangelog] = mockChangelog;
 
 describe("Bounce Rule Detailed", () => {
-  let wrapper;
+  let props;
+  let mountedDetailsPage;
+  const DetailsPage = () => {
+    if (!mountedDetailsPage) {
+      mountedDetailsPage = shallow(<BounceRuleDetailed {...props} />);
+    }
+    return mountedDetailsPage;
+  };
   beforeEach(() => {
-    wrapper = shallow(BounceRuleDetailedObject);
+    props = {
+      currentRule: sampleRule,
+      changelog: [sampleChangelog],
+      filteredChangelog: [sampleChangelog],
+    };
   });
 
   it("should render correctly", () => {
     expect(
-      renderer.create(<StaticRouter>{BounceRuleDetailedObject}</StaticRouter>)
+      renderer.create(
+        <Router>
+          <BounceRuleDetailed {...props} />
+        </Router>
+      )
     ).toMatchSnapshot();
   });
 
-  it("should render details table", () => {
-    expect(wrapper.find(details).exists()).toBeTruthy();
-  });
+  describe("When a user visits the detailed rule page", () => {
+    it("should render details table", () => {
+      expect(DetailsPage().find(details)).toHaveLength(1);
+    });
 
-  it("should render change log table", () => {
-    expect(wrapper.find(changelog).exists()).toBeTruthy();
-  });
+    it("should render change log table", () => {
+      expect(DetailsPage().find(changelog)).toHaveLength(1);
+    });
 
-  it("should show cancel/save button when edit is clicked", () => {
-    expect(wrapper.find(editButton).exists()).toBeTruthy();
-    wrapper.setProps({
-      isEditClicked: true,
+    it("should show edit button", () => {
+      expect(DetailsPage().find(editButton)).toHaveLength(1);
     });
-    expect(wrapper.find(cancelButton).exists()).toBeTruthy();
-    expect(wrapper.find(saveButton).exists()).toBeTruthy();
-  });
 
-  it("should be editable when edit is clicked", () => {
-    wrapper.setProps({
-      isEditClicked: true,
+    it("should render paginiation", () => {
+      expect(DetailsPage().find(pagination)).toHaveLength(1);
```

Comment:
can also do .find().exists()).toBe(true). No need to change anything cause this works too but giving another option for other tests 😄